src/main/java/manager/DeadlineManager.java

```diff
@@ -0,0 +1,118 @@
+     * @param ui       manages user output.
+     */
+    public static void addDeadline(Deadline deadline, TextUi ui) {
+        int len = deadlines.size();
+        deadlines.add(deadline);
+        //Solution below adapted from https://github.com/darrenangwx/ip/blob/6d3f1bc5f1
+        // a281f9459a67650b043705d3096a8f/src/main/java/task/TaskParser.java
+        ui.printMessage(Messages.MESSAGE_DEADLINE_ADDED +
+                deadlines.get(deadlines.size() - 1).toString() +
+                String.format(Messages.MESSAGE_NUMBER_OF_DEADLINES, deadlines.size()));
+        assert deadlines.size() == len + 1 : "Length of deadline list should increase by 1";
+        try {
+            DeadlineStorage deadlineStorage = new DeadlineStorage();
+            deadlineStorage.writeToDeadlineFile(deadlines);
+        } catch (IOException e) {
+            ui.printMessage(String.format(Messages.ERROR_STORAGE_INVALID_WRITE_LINE, deadline));
+        }
+    }
+
+    /**
+     * Print the deadline list.
+     *
+     * @param ui manages user output.
+     */
+    public static void printDeadlines(TextUi ui) {
+        try {
+            if (deadlines.isEmpty()) {
+                throw new DinerDirectorException(Messages.MESSAGE_DEADLINE_EMPTY_LIST);
+            }
+            System.out.println(Messages.MESSAGE_DEADLINE_VIEW_LIST);
+            for (int i = 1; i <= deadlines.size(); i++) {
+                ui.printMessage(i + ". " + deadlines.get(i - 1).toString());
+            }
+        } catch (DinerDirectorException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Deletes a deadline from the deadline list.
+     *
+     * @param index index of deadline to be deleted.
+     * @param ui    manages user output.
+     */
+    public static void deleteDeadline(int index, TextUi ui) {
+        int len = deadlines.size();
+        try {
+            System.out.print(Messages.MESSAGE_DEADLINE_REMOVED + deadlines.get(index).toString());
+            deadlines.remove(index);
+            ui.printMessage(String.format(Messages.MESSAGE_NUMBER_OF_DEADLINES, deadlines.size()));
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(Messages.ERROR_DEADLINE_INVALID_INDEX);
+            return;
+        }
+        assert deadlines.size() == len - 1 : "Length of deadline list should decrease by 1.";
+        try {
+            DeadlineStorage deadlineStorage = new DeadlineStorage();
+            deadlineStorage.writeToDeadlineFile(deadlines);
+        } catch (IOException e) {
+            ui.printMessage(Messages.ERROR_STORAGE_DELETE_FAILED);
+        }
+    }
+
+    /**
+     * Prints a list of deadline that matches the keyword.
+     *
+     * @param keyword the keyword to search for.
+     * @param ui manages user output.
+     */
+    public static void findDeadline(String keyword, TextUi ui) {
+        ArrayList<Deadline> matchingDeadlines = new ArrayList<>();
+        for (Deadline x : deadlines) {
```

Comment:
@Stella1585 
default is use the same character as the object name, for example Deadline d